testutils/src/main/scala/za/co/absa/enceladus/testutils/e2eSparkRunner/CmdConfig.scala

```diff
@@ -38,10 +39,74 @@ case class CmdConfig(rawFormat: String = "xml",
                      datasetVersion: String = "",
                      reportDate: String = "",
                      reportVersion: String = "",
-                     sparkConfFile: String = "")
+                     sparkConfFile: String = "") {
+  private val stdPaths = "conformance-output/standardized-" +
+                        s"$datasetName-" +
+                        s"$datasetVersion-" +
+                        s"$reportDate-" +
+                        s"$reportVersion"
+  private val confPaths = s"$datasetName/" +
+                          s"enceladus_info_date=$reportDate/" +
+                          s"enceladus_info_version=$reportVersion"
+
+  private val defaultStdOut = s"/tmp/$stdPaths"
+  private val defaultConfOut = s"/publish/$confPaths"
+
+  private val refStdOut = s"/ref/tmp/$stdPaths"
+  private val refConfOut = s"/ref/publish/$confPaths"
+
+  private val cmpStdPath = s"/cmp/tmp/$stdPaths"
+  private val cmpConfPath = s"/cmp/publish/$confPaths"
+
+  val sparkConf: String = Source.fromFile(sparkConfFile).getLines().mkString
+
+  val stdComparisonConf: String = comparisonConf(defaultStdOut, refStdOut, cmpStdPath, keys)
+  val confComparisonConf: String = comparisonConf(defaultConfOut, refConfOut, cmpConfPath, keys)
+
+  /**
+    * Get Conformance options like dataset name, dataset version, etc.
+    * @return A string of conformance job options
+    */
+  def conformanceJobConf: String = {
+    s"--dataset-name $datasetName " +
+      s"--dataset-version $datasetVersion " +
+      s"--report-date $reportDate " +
+      s"--report-version $reportVersion " +
+      s"--menas-credentials-file $menasCredentialsFile"
+  }
+
+  /**
+    * Get Standartization options like dataset name, dataset version, etc.
+    * @return A string of Standartization job options
+    */
+  def standartizationJobConf: String = {
+    val base: String = s"--dataset-name $datasetName " +
+      s"--dataset-version $datasetVersion " +
+      s"--report-date $reportDate " +
+      s"--report-version $reportVersion " +
+      s"--menas-credentials-file $menasCredentialsFile"
+
+    val format: String = rawFormat match {
+      case "csv" => s"--raw-format $rawFormat --delimiter ${csvDelimiter.get} --header ${csvHeader.get}"
```

Comment:
I wouldn't use `.get` since it is valid to leave them out for csv as they have default values.

Reply:
+1. You use `.get` in several places. Consider changing to pattern matching where applicable.
(This may be added to a discussion as a very often occurred anti-pattern).

Instead of
```scala
if (opt.isEmpty) {
 // Something
} else {
  doSomething(opt.get)
}
```
use
```scala
opt match {
  case Some(v) => doSomething(v)
  case None => // Something
}
```
The latter if more resillient to accidental throw if the code evolves eventually.

Reply:
I would rather change the scopt `--help` of it saying default value is X and let the user of this script deal with it however he wants. This e2e runner is for predefined tests only and it has expectations, one of them being that fields are filled at full and correctly or the tester is happy with the default.

As for the `if else`, I don't see it anywhere. I am just plainly using `get` for the values. The reason why these parameters are an `Option` is mostly that it was done so, in all the previous `CmdConfig` cases and I wanted to make it easier for the person that will in future be adding something.

Reply:
Indeed, `scopt` forces to use `.get` a lot. I'm fine with that.

Sorry about the anti-pattern. I see only one place where this is applicable. No explicit `else`, but still...
```scala
    if (optionCmd.isEmpty) {
      System.exit(1)
    }
    optionCmd.get
```

Reply:
That is something I got from standartization and conformance cmd configs. I can look into it and re-do it. Do you want me to re-do it in other cases aswell or just mine?

Reply:
Just here. Currently, David and Jan are doing things in Conformance. And I plan to start working on it next week. Better to avoid changes there for now to decrease chances of hard merge conflicts.

Reply:
The reason `System.exit(1)` is done this way is because the method requires a return value which `System::exit` does not provide. You can either:
1. put a useless return value to please the compiler after the `System::exit` call (very hacky)
2. change the return type to `Option` and deal with the `None` value in the calling function (a more appropriate place to `System::exit`, I think)

Reply:
> This e2e runner is for predefined tests only and it has expectations, one of them being that fields are filled at full and correctly or the tester is happy with the default.

The problem is the default will fail with `None.get` even if the tester is happy with it, since the default is not generated by `scopt` but assumed by `spark`.

Reply:
Oh, I see. I'm fine with leaving it as is or go with option 2. Up to you guys.

Reply:
> The problem is the default will fail with `None.get` even if the tester is happy with it, since the default is not generated by `scopt` but assumed by `spark`.

Shouldn't it, if it's optional and has a default value, keep the default value? I am unable to test this hypothesis right now.

Reply:
So I had some issues on my local machine I fixed. The default value is working if the parameter is optional. Which means it performs as intended. I would get rid of the option but I read about it being needed for optional params.

Reply:
My bad, didn't see the case class's defaults were set, I was under the impression they were left as `None`.